lib/rexml/parsers/xpathparser.rb

```diff
@@ -188,6 +188,21 @@ def predicate_to_string( path, &block )
         return string.squeeze(" ")
       end
 
+      def quote_literal( literal )
+        case literal
+        when String
+          # XPath 1.0 does not support escape characters.
+          # Assumes literal does not contain both single and double quotes.
+          if literal.include?("'")
+            "\"#{literal}\""
+          else
+            "'#{literal}'"
+          end
+        else
+          literal.inspect
+        end
+      end
+
       private
```

Comment:
Could you move this `private` above `quote_literal`?

Reply:
Done: d6a02b91a244b8c0805a780525568ad4bc2b74de